### ISSUE TITLE
Fetch klines of multiple symbols

### ISSUE THREAD

**Issue by hemulin** [bullsignals/binance-downloader#14]
Hi,
That is really a nice project. Well done.

Is it possible to add functionality of downloading klines for multiple symbols at once?
Either by passing an array of symbols or perhaps some criteria such as top-x symbols by marketcap.

How much do you think Binance servers won't like such requests?

**Comment by samuelhei** [comment]
Thanks! Good Suggestion.

There is a rate limit of 1200 requests by minute (10000k rows by second), but the app is not requesting in parallel (for now) so it is much slower, so we are fine.

**Comment by samuelhei** [comment]
One possible implementation for this is the following:

Turn the  `--symbol SYMBOL, -s SYMBOL` parameter optional and then create another parameter also optional `--quantity QUANTITY, -q QUANTITY` that gets the total number of coins (default 1) and then get the first QUANTITY symbols on this endpoint [https://api.binance.com/api/v1/ticker/24hr](https://api.binance.com/api/v1/ticker/24hr)

**Comment by samuelhei** [comment]
This one is also a good issue @santoshyadav198613 

Let me know if you are up for it ;)

**Comment by anderson89marques** [comment]
Hi @samuelhei, 
How about we specify what symbols we want instead of quantifying? 
So we could use `--symbols Symbol1 Symbol2` and the default would be the same.
Another question:
with multiple symbols, we will have multiple files, right? 
or we should put all symbols in one single file?
for me, the first option is the right answer.
What do you think about?

**Comment by kenorb** [comment]
We can separate by a comma. Check the [similar script](https://github.com/FX31337/FX-BT-Scripts/blob/576901de03459541f8730836aa7d61d67ddffd38/dl_bt_dukascopy.py#L379).

    parser.add_argument("-p", "--pairs",        action="store",       dest="pairs",     help="Pair(s) to download (separated by comma).", default="all")
    ...
    pairs =  list(all_currencies.keys()) if args.pairs  == "all" else args.pairs.split(',')
    ...
        for pair in sorted(pairs):

**Comment by samuelhei** [comment]
> with multiple symbols, we will have multiple files, right?

Yes, @anderson89marques  different symbols we can save in different files.

> We can separate by a comma. Check the similar script.

Good idea @kenorb, I will check out the script

**Comment by evanatyourservice** [comment]
I just used bash for now, save below as a .sh file (including the done on the bottom), then run
bash filename.sh
in command line.
 
#!/bin/bash
CoinList=("BTCUSDT ETHUSDT LTCUSDT BNBUSDT NEOUSDT ETHBTC LTCBTC BNBBTC EOSBTC")
Start="2018/01/01"
End="2018/01/02"
Output="./binance/train/timestamp_symbol_interval"
for coin in $CoinList
do kline-binance --start $Start --end $End $coin 1m
done

**Comment by ahmedhassanismail** [comment]
Hello,

at the first it's good utility. thanks for you effort to build. 

I need help to add new columns into the csv files output by symbol and interval flag like the below example:

OpenTime,Open,High,Low,Close,Volume,CloseTime,qav,numTrades,tbbav,tbqav,ignore,++symbol,interval 

Thank you in advance.
